jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -257,21 +266,19 @@ object XGBoost extends Serializable {
    * @param useExternalMemory indicate whether to use external memory cache, by setting this flag as
    *                          true, the user may save the RAM cost for running XGBoost within Spark
    * @param missing the value represented the missing value in the dataset
-   * @param baseMargin initial prediction for boosting.
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
   @throws(classOf[XGBoostError])
   def trainWithRDD(
-      trainingData: RDD[MLLabeledPoint],
+      trainingData: RDD[XGBLabeledPoint],
```

Comment:
> Make XGBoost.trainWithRDD private and keep baseMargin there.

No, I mean do not modify the signature of RDD-relevant APIs, and only support baseMargin in DataFrame API. 

actually my thought is 

1. we have new features to add in, 

2. to add this feature, we need to do break backward-compatibility, or only add in part of APIs

This issue also appears when Spark moving from mllib to ml.

Since Spark community has supported new features only in ML, and the major reason for XGBoost4J-Spark to exist is to offer a seamless integration solution for Spark, I think it is fine to only support new features in DataFrame API